Parsely/wp-parsely
# Improve handling the "Force HTTPS canonicals" option

Close #174 

## Description

There are two changes on this PR:

1, The default value for `Force HTTPS canonicals` option is set based on the protocol of `home_url()` value: 
- true for https
- false for HTTP

2, The help text for `Force HTTPS canonicals` is changed but generally, it does not intensify that users will need to change this value. They just need to do that if necessary. 

**NOTE: there is a slight change for (2) comparing with this comment https://github.com/Parsely/wp-parsely/issues/174#issuecomment-831762629 as I do not want to let all existing users contact Parse.ly support rep to run re-indexing. That might create unnecessary requests for Parse.ly support team.**

## Motivation and Context

See #174. 

## How Has This Been Tested?

#### Test 1: A new site with HTTP address

1. Delete `parsely` option via wp-cli: `wp option delete parsely` or directly in the database.
2. Visit `wp-admin/options-general.php?page=parsely`, `Force HTTPS canonicals` is set to `No`.
3. Change ``Force HTTPS canonicals` to `Yes` or change the site address to HTTPS. 
4. See this message `This option is not being consistent with the protocol of your site address. You can keep it as-is without affecting your Parse.ly data. However, if you're forced to change this value, please reach out to Parse.ly support`

#### Test 2: Test a new site with HTTPS address

1. Similar step 1 in test 1.
2. Visit `wp-admin/options-general.php?page=parsely`, `Force HTTPS canonicals` is set to `Yes`.
3. Change `Force HTTPS canonicals` to `No` or change the site address to HTTP. 
4. Similar to step 4 in test 1. 

#### Test 3: Test with a site upgrading from the previous version to this version. 

1. Use Parsely 2.4.1 - the current latest stable version.
2. Visit `wp-admin/options-general.php?page=parsely`, choose whatever value for `Force HTTPS canonicals`.
3. Change the plugin to this PR. 
4. Visit `wp-admin/options-general.php?page=parsely`, the value chosen in step 2 has not been changed.

## Screenshots (if appropriate):

## Types of changes


Note: I added Improvement because all 3 previous options seem not to make sense for this PR.

## Consideration 

IMHO, this is pretty subtle and we do not need tests. But in case, any reviewer thinks tests are necessary, I can add them in new commits.
